### PR TITLE
Move `pkg/operation/botanist/component` to `pkg/component`

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:
The `operation/botanist` packages are very specific to `gardenlet`'s `Seed/Shoot` controller, however the components are reused also in other places (e.g. `gardener-operator`'s `Garden` controller).

**Release note**:

```breaking developer
The `pkg/operation/botanist/component/*` resources have been moved to `pkg/component/*`.
```

### CODE REVIEW

File: hack/check-license-header.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ missing_license_header_files="$(addlicense \
   -ignore "**/*.yaml" \
   -ignore "**/Dockerfile" \
   -ignore "hack/tools/gomegacheck/**" \
-  -ignore "pkg/operation/botanist/**/*.sh" \
+  -ignore "pkg/component/**/*.sh" \
```
Comment (acumino) [inline hack/check-license-header.sh:31]: Same change in file `hack/add-license-header.sh`.

File: pkg/component/kubeapiserver/deployment.go [hunk 2]
```diff
@@ -482,7 +482,7 @@ func (k *kubeAPIServer) reconcileDeployment(
 			// For kube-apiserver version 1.24 there is a deadlock that can occur during shutdown that prevents the
 			// graceful termination of the kube-apiserver container to complete when the --audit-log-mode setting
 			// is set to batch. For more information check
-			// https://github.com/gardener/gardener/blob/a63e23a27dabc6a25fb470128a52f8585cd136ff/pkg/operation/botanist/component/kubeapiserver/deployment.go#L677-L683
+			// https://github.com/gardener/gardener/blob/a63e23a27dabc6a25fb470128a52f8585cd136ff/pkg/component/kubeapiserver/deployment.go#L677-L683
```
Comment (acumino) [inline pkg/component/kubeapiserver/deployment.go]: This is a dead link.

**Review by acumino — COMMENTED** [review]
Few nits.

**Review by acumino — COMMENTED** [review]
/lgtm

**Review by timuthy — COMMENTED** [review]
/lgtm

**Pull request merged**